### PR TITLE
Implement the DSL for commentFilePath to fix #514

### LINKED ISSUE #514: commentFilePath missing from DSL
It appears that when the comment file option was added to the plugin, it was never added to the DSL. I have yet to figure out how to add it via. groovy, so is it possible to get this added to the DSL? This option is also missing from the global settings for the plugin, so we can't even add a default for all jobs on the server.

In the meantime, does anyone have any suggestions for adding this via. groovy? I've been trying various variations on the following with no success:
```groovy
          configure { node ->
              def extensions = node / 'triggers' / 'org.jenkinsci.plugins.ghprb.GhprbTrigger'('plugin': 'ghprb@1.36.2') / 'extensions'
              extensions << 'org.jenkinsci.plugins.ghprb.extensions.comments.GhprbCommentFile' {
                  commentFilePath('github-comments.txt')
              }
          }
```

### PR DESCRIPTION
**Pull request by hibes**

### CODE REVIEW

**Comment by hibes** [comment]
@samrocketman, had some issues with the other PR.  This one has the collection of changes and is up to date with master.

**Comment by Kentalot** [comment]
Maybe make a seed job and have it try to process a dsl with this option on?

**Comment by samrocketman** [comment]
I don't have time to work on testing this at the moment since I'm preparing for Jenkins World next week where I'm presenting.

File: README.md [hunk 1]
```diff
@@ -144,6 +144,7 @@ job('upstreamJob') {
             permitAll()
             autoCloseFailedPullRequests()
             displayBuildErrorsOnDownstreamBuilds()
+            commentFilePath("relative/path/to/file")
```
Comment (Kentalot) [inline README.md]: Actually, isn't this changed now according to this: https://github.com/jenkinsci/ghprb-plugin/pull/637#issuecomment-388125458 

@bjoernhaeuser ?

**Comment by samrocketman** [comment]
Yeah, the job dsl testing happens manually at this time.

**Comment by hibes** [comment]
@samrocketman 

Merging is blocked
Merging can be performed automatically with 1 approving review.

I pushed the README update requested by @Kentalot.

**Comment by Kentalot** [comment]
@samrocketman Any updates on this one? 👍

**Comment by samrocketman** [comment]
Not available for coding.  Traveling on vacation at the moment.

**Comment by samrocketman** [comment]
I will be back home Oct 17th.  If you see no action in the week following that date feel free to ping me again to remind me.

**Comment by samrocketman** [comment]
@bjoernhaeuser feel free to merge LGTM

**Comment by bjoernhaeuser** [comment]
@samrocketman thanks for approving.

**Comment by Kentalot** [comment]
Awesome now that this is merged, any way we could get https://github.com/jenkinsci/ghprb-plugin/pull/682 merged? :)

**Comment by Kentalot** [comment]
@bjoernhaeuser I tested this on my DSL job and it seems to not work as it somehow does not seem to find commentFilePath method that takes in a closure.  Not sure what's happening.  Your #664 update also fails with a similar error.  I checked to make sure i have the latest ghprb plugin version (1.42.0). commitStatus and buildStatus extensions work just fine.  I don't see anything wrong with your code and the only difference I could see between the working ones (commitStatus and buildStatus) and the non-working ones (commentFilePath and cancelBuildsOnUpdate) is the latter ones have their methods public.  I doubt that's the issue, but I couldn't see any other difference.

Here's the error:
```
Processing DSL script PipelineJob.groovy
ERROR: (PullRequestTrigger.groovy, line 110) No signature of method: javaposse.jobdsl.dsl.helpers.triggers.TriggerContext.cancelBuildsOnUpdate() is applicable for argument types: (jobs.utils.builders.pipelines.scripts.PullRequestTrigger$_addTriggerTo_closure1$_closure2$_closure3$_closure6) values: [jobs.utils.builders.pipelines.scripts.PullRequestTrigger$_addTriggerTo_closure1$_closure2$_closure3$_closure6@6052ddc2]
```
and the code snippet:
```
extensions {
                    commitStatus {
                        context(gitStatus.toString())
                        triggeredStatus('Triggering Jenkins...')
                        startedStatus("Running $gitStatus on Jenkins...")
                        addTestResults(true)
                        completedStatus('SUCCESS', 'Tests passed')
                        completedStatus('FAILURE', 'Tests failed')
                        completedStatus('PENDING', 'Testing in progress...')
                        completedStatus('ERROR', 'Pipeline crashed')
                    }
                    buildStatus {
                        completedStatus('SUCCESS', successMessage)
                        completedStatus('FAILURE', failureMessage)
                        completedStatus('ERROR',
                                'There was an error in the infrastructure, please contact the Jenkins maintainer.')
                    }
                    cancelBuildsOnUpdate {
                        overrideGlobal(true)
                    }
                    commentFilePath {
                            commentFilePath(extraCommentFilePath)
                    }                    
                }
```

**Pull request merged**